lib/model/folder.go

```diff
@@ -127,24 +136,60 @@ func (f *folder) scanTimerFired() {
 	f.scan.Reschedule()
 }
 
+func (f *folder) WatchError() error {
+	f.watchErrMut.Lock()
+	defer f.watchErrMut.Unlock()
+	return f.watchErr
+}
+
 func (f *folder) startWatch() {
 	ctx, cancel := context.WithCancel(f.ctx)
 	f.model.fmut.RLock()
 	ignores := f.model.folderIgnores[f.folderID]
 	f.model.fmut.RUnlock()
-	eventChan, err := f.Filesystem().Watch(".", ignores, ctx, f.IgnorePerms)
-	if err != nil {
-		l.Warnf("Failed to start filesystem watcher for folder %s: %v", f.Description(), err)
-	} else {
-		f.watchChan = make(chan []string)
-		f.watchCancel = cancel
-		watchaggregator.Aggregate(eventChan, f.watchChan, f.FolderConfiguration, f.model.cfg, ctx)
-		l.Infoln("Started filesystem watcher for folder", f.Description())
+	f.watchChan = make(chan []string)
```

Comment:
When I look at it now I'm a bit suspicious that setting these without locking might be a data race. I'll try to prove it one way or the other...

Reply:
I can't tickle the race detector to complain about it, so I guess it's fine

Reply:
`startWatch` is only ever to be called synchronously from the main for loop in the folder's `Serve` - that's why `scheduleWatchRestart` exists and probably deserves a few comments.